mise.toml

```diff
@@ -17,8 +17,8 @@ svm-rs = "0.5.19"
 "go:github.com/ethereum/go-ethereum/cmd/abigen" = "1.15.10"
 "go:github.com/ethereum/go-ethereum/cmd/geth" = "1.16.4" # Osaka testnet release.
 "go:gotest.tools/gotestsum" = "1.12.1"
-"go:github.com/vektra/mockery/v2" = "2.46.0"
-"go:github.com/golangci/golangci-lint/cmd/golangci-lint" = "1.64.8"
+"go:github.com/vektra/mockery/v2" = "2.53"
+"go:github.com/golangci/golangci-lint/v2/cmd/golangci-lint" = "v2.4.0"
```

Comment:
ah, I think this is where all the changes are coming from. Funnily enough, we don't need to upgrade to v2, we just need to ensure the current version gets rebuilt with go1.24. If you delete where mise has already instatlled it and run mise install again lint passes cleanly.

Upgrading to v2 is probably a good idea but this many changes alongside a go and op-geth upgrade makes me very nervous (and rather breaks my browser in GitHub's review view). We could also just start by disabling the new rules its adding and then enable them more individually (if they actually add value).